app/src/components/features/rules/RuleBuilder/RuleBuilder.js

```diff
@@ -47,7 +47,9 @@ const { RULE_EDITOR_CONFIG, RULE_TYPES_CONFIG } = APP_CONSTANTS;
 const RuleBuilder = (props) => {
   const location = useLocation();
   const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
   const { state } = location;
+  const ruleGroupId = searchParams.get("groupId") ?? "";
```

Comment:
What happens if a user enters a wrong `groupId` in url? Does the group gets created or rule get created on undefined group?